### PR TITLE
Add release branch script

### PR DESCRIPTION
**Pull request by axue-broadcom**
This PR adds a script that allows developers to easily create a release branch for a particular major.minor version, update the version file and tag any associated images needed for the release pipeline. This script also prints out the necessary make command to set the release pipeline for the branch.

### CODE REVIEW

File: dev/create-release-branch.bash [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      ${cmd_to_run}
+      break ;;
+    [Nn]*) break ;;
+    *) echo "Please answer yes or no." ;;
+    esac
+  done
+}
+
+function create_branch_and_snapshot_commit() {
+  git checkout "${GIT_SHA}"
```
Comment (denalex) [inline dev/create-release-branch.bash:79]: Do we need `-C "${pxf_src}"` here and on line 81 ?

File: dev/create-release-branch.bash [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+  printf "Changing version to ${SNAPSHOT_VERSION} and committing change..."
+  printf "${SNAPSHOT_VERSION}" > "${pxf_src}"/version
+  git -C "${pxf_src}" add version
+  git -C "${pxf_src}" commit -m "Bump version to ${SNAPSHOT_VERSION} [skip ci]"
+
+  printf "Pushing new branch ${branch_name} and new SNAPSHOT version ${SNAPSHOT_VERSION} to ${GIT_REMOTE}"
+  git -C "${pxf_src}" push -u "${GIT_REMOTE}" "${branch_name}"
+
+  git checkout "${current_branch}"
```
Comment (denalex) [inline dev/create-release-branch.bash:94]: should we remove the local intermediate branch name `${branch_name}` ?

File: dev/create-release-branch.bash [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+# get the version from the branch name
+branch_name=branch-"${MAJOR_MINOR_VERSION}".x
+current_branch=$(git rev-parse --abbrev-ref HEAD)
+image_tag=release-"${MAJOR_MINOR_VERSION}".x
+
+# double check that they want to create the branch
+prompt_for_confirmation "Create branch \`${branch_name}\` on git remote \`${GIT_REMOTE}\` and push the initial SNAPSHOT commit (y/n)? " create_branch_and_snapshot_commit
+
+# create the new tag for the branches
+prompt_for_confirmation "Create tags \`${image_tag}\` for branch \`${branch_name}\` (y/n)? " tag_images
```
Comment (denalex) [inline dev/create-release-branch.bash:158]:
```suggestion
prompt_for_confirmation "Create docker tag \`${image_tag}\` for images to use with branch \`${branch_name}\` (y/n)? " tag_images
```

File: dev/create-release-branch.bash [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  'gpdb-pxf-dev/gpdb6-oel7-test-pxf'
+)
+
+RPMREBUILD_IMAGE_LIST=(
+  'rpmrebuild-centos7'
+  'rpmrebuild-rocky8'
+  'rpmrebuild-rocky9'
+)
+
+  echo "Tagging current latest images with \`${image_tag}\`"
```
Comment (denalex) [inline dev/create-release-branch.bash:135]:
```suggestion
  echo "Tagging current latest docker images with \`${image_tag}\`"
```

File: dev/create-release-branch.bash [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+image_tag=release-"${MAJOR_MINOR_VERSION}".x
+
+# double check that they want to create the branch
+prompt_for_confirmation "Create branch \`${branch_name}\` on git remote \`${GIT_REMOTE}\` and push the initial SNAPSHOT commit (y/n)? " create_branch_and_snapshot_commit
+
+# create the new tag for the branches
+prompt_for_confirmation "Create tags \`${image_tag}\` for branch \`${branch_name}\` (y/n)? " tag_images
+
+printf "\nPlease set the release pipeline using:\n\n"
+printf '\tmake release RELEASE_BRANCH=%s RELEASE_IMAGE_TAG=%s\n\n' "${branch_name}" "${image_tag}"
```
Comment (denalex) [inline dev/create-release-branch.bash:161]: there should be convention for deriving RELEASE_IMAGE_TAG from RELEASE_BRANCH, so we don't have to provide it as a parameter. we can still have a capacity to specify and override, but that should not be a default model, IMO.

File: dev/create-release-branch.bash [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+}
+
+# --- main script logic ---
+
+# do basic validation
+if [ $# -lt 2 ]; then
+  print_usage
+  exit 1
+fi
+validate_version
```
[the remark this answers is not in the document]
Comment (axue-broadcom) [reply dev/create-release-branch.bash:147]: there are confirmations for this. you can choose to skip branch creation again and just push the images if needed. but i can add a validation check for git and gcloud

File: dev/create-release-branch.bash (hunk 2 above)
Comment (axue-broadcom) [reply dev/create-release-branch.bash:94]: `branch_name` is the name of the new release branch

File: dev/create-release-branch.bash (hunk 1 above)
[the remark this answers is not in the document]
Comment (axue-broadcom) [reply dev/create-release-branch.bash]: I have added a check that will cause the script to error out if they are not on main

Comment (axue-broadcom) [reply dev/create-release-branch.bash]: technically no, it was being used for updating and editing the version file, I have modified how tis is done so `-C` is not needed for any of the git commands

File: concourse/Makefile [hunk 7]
```diff
@@ -68,12 +69,13 @@ set-release-pipeline:
 ifndef RELEASE_BRANCH
 	$(error RELEASE_BRANCH is not set, please set a release branch)
 else
+RELEASE_IMAGE_TAG = release-$(word 2, $(subst -, , $(GEMFIRE_VERSION)))
```
Comment (yimingli-vmware) [inline concourse/Makefile]:
```suggestion
RELEASE_IMAGE_TAG = release-$(word 2, $(subst -, , $(PXF_VERSION)))
```

File: dev/create-release-branch.bash [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+      ${cmd_to_run}
+      break ;;
+    [Nn]*) break ;;
+    *) echo "Please answer yes or no." ;;
+    esac
+  done
+}
+
+function create_branch_and_snapshot_commit() {
+  printf "Creating new branch %s...\n" "${branch_name}"
```
Comment (yimingli-vmware) [inline dev/create-release-branch.bash:76]: let's use `echo`

File: dev/create-release-branch.bash [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+
+# double check that they want to create the branch
+prompt_for_confirmation "Create branch \`${branch_name}\` on git remote \`${GIT_REMOTE}\` and push the initial SNAPSHOT commit (y/n)? " create_branch_and_snapshot_commit
+
+# create the new tag for the branches
+prompt_for_confirmation "Create docker tag \`${image_tag}\` for images to use with the release pipeline for branch \`${branch_name}\` (y/n)? " tag_images
+
+echo "Please set the release pipeline using:"
+echo ""
+echo "    make -C ${pxf_src}/concourse release RELEASE_BRANCH=${branch_name}"
```
Comment (denalex) [inline dev/create-release-branch.bash:165]: `make release` target gives me a feeling we will make a release, but in reality we will only fly a pipeline for a non-dev branch. We can change it later, just mentioning it here.

**Review by denalex — APPROVED** [review]

File: dev/create-release-branch.bash [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+
+    would first ask for confirmation on creating the release branch. After which it would
+    create a branch for the release called \`branch-1.2.x\` off commit \`abcdef\`.
+    It would also update the version inside pxf/version to be \`1.2.0-SNAPSHOT\`.
+
+    Then, it would ask for confirmation on tagging the images needed for the release branch.
+    After confirmation, this script would update the necessary images with \`release-1.2.x\`
+    and then print out the make command for setting the new release pipeline to use the newly
+    created branch and the newly tagged images.
+)
```
Comment (djianwen-vmware) [inline dev/create-release-branch.bash:40]: What is this close parenthesis used for?

**Review by yimingli-vmware — APPROVED** [review]

**Review by tatb-vmware — APPROVED** [review]

**Pull request merged**